Ubuntu/kinetic: release 23.1-0ubuntu0~22.10.1

aciba90 opened the pull request:
# DO NOT SQUASH

## Steps to reproduce

```sh
git fetch upstream --tags
git checkout -B upstream/ubuntu/kinetic ubuntu/kinetic
new_upstream_snapshot.py -c 23.1 -b 2008230
```

## Checks

- [x] build-package
- [x] sbuild
- [x] no lintian errors

TheRealFalcon commented:
Main difference I see is in quilt refresh:
```diff
diff --git a/debian/changelog b/debian/changelog
index a3a5bf8f2..d199d299e 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -7,7 +7,7 @@ cloud-init (23.1-0ubuntu0~22.10.1) kinetic; urgency=medium
     List of changes from upstream can be found at
     https://raw.githubusercontent.com/canonical/cloud-init/23.1/ChangeLog
 
- -- Alberto Contreras <alberto.contreras@canonical.com>  Thu, 23 Feb 2023 12:32:59 +0100
+ -- James Falcon <james.falcon@canonical.com>  Thu, 23 Feb 2023 10:15:20 -0600
 
 cloud-init (22.4-0ubuntu0~22.10.1) kinetic; urgency=medium
 
diff --git a/debian/patches/retain-netplan-world-readable.patch b/debian/patches/retain-netplan-world-readable.patch
index eff11bd64..2e206a41d 100644
--- a/debian/patches/retain-netplan-world-readable.patch
+++ b/debian/patches/retain-netplan-world-readable.patch
@@ -7,10 +7,8 @@ Forwarded: not-needed
 Last-Update: 2023-01-09 
 ---
 This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
-Index: cloud-init/cloudinit/features.py
-===================================================================
---- cloud-init.orig/cloudinit/features.py
-+++ cloud-init/cloudinit/features.py
+--- a/cloudinit/features.py
++++ b/cloudinit/features.py
 @@ -59,7 +59,7 @@ only non-hashed passwords were expired.
  (This flag can be removed after Jammy is no longer supported.)
  """
@@ -20,10 +18,8 @@ Index: cloud-init/cloudinit/features.py
  """
  If ``NETPLAN_CONFIG_ROOT_READ_ONLY`` is True, then netplan configuration will
  be written as a single root readon-only file /etc/netplan/50-cloud-init.yaml.
-Index: cloud-init/tests/unittests/distros/test_netconfig.py
-===================================================================
---- cloud-init.orig/tests/unittests/distros/test_netconfig.py
-+++ cloud-init/tests/unittests/distros/test_netconfig.py
+--- a/tests/unittests/distros/test_netconfig.py
++++ b/tests/unittests/distros/test_netconfig.py
 @@ -592,32 +592,41 @@ class TestNetCfgDistroUbuntuNetplan(Test
              (self.netplan_path(), V1_TO_V2_NET_CFG_OUTPUT, 0o600),
          )

```

I would think that the QUILT options at the top of `new_upstream_snapshot.py` (particularly `-p ab`) should make this consistent.

blackboxsw commented on debian/changelog:
```diff
@@ -1,10 +1,13 @@
-cloud-init (22.4-0ubuntu0~22.10.2) UNRELEASED; urgency=medium
+cloud-init (23.1-0ubuntu0~22.10.1) kinetic; urgency=medium
```
What we need to add to release documents (and be aware of). Is whether or not our previous release 22.4 had a hot-fix release branch `upstream/ubuntu/kinetic-22.4.x`. If it did, we need to manually sync the top-most debian/changelog entries which don't exist yet in `upstream/ubuntu/devel` which were already officially published. If we don't sync those released changelog entries our diff looks confusing to SRU reviewers (and essentially redacts content that was already publicly visible for known releases).

So, when SRU-ing a new release, prior to new_upstream_snapshot.py, we:
- [ ] check for the presence of a hot fix branch for the previous release `upstream/ubuntu/<SERIES>-22,4,x` 
- [ ] validate any d/changelog and manually sync it `git diff upstream/ubuntu/<SERIES>-22.4.x debian/changelog`
- [ ] validate any Changelog and manually sync it `git diff upstream/ubuntu/<SERIES>-22.4.x Changelog`

Manually add the missing entry either via git cherry-pick or cut-n-paste (watch for trailing whitespace).  git commit -m 'sync d/changelog from hot fix branch ubuntu/kinetic-22.4.x` debian/changelog. Then git rebase -i HEAD~3 and place that commit be

There is a diff telling us we are missing `cloud-init (22.4.2-0ubuntu0~22.10.1) kinetic; urgency=medium` top-most changelog section
```diff
csmith@downtown:/tmp/cloud-init ((d536a5a1...))$ git diff upstream/ubuntu/kinetic-22.4.x debian/changelog 
diff --git a/debian/changelog b/debian/changelog
index c5368242..a3a5bf8f 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,11 +1,13 @@
-cloud-init (22.4.2-0ubuntu0~22.10.1) kinetic; urgency=medium
+cloud-init (23.1-0ubuntu0~22.10.1) kinetic; urgency=medium
 
-  * Upstream snapshot based on 22.4.2 upstream release. (LP: #1996645)
-    - List of changes from upstream can be found at
-      https://raw.githubusercontent.com/canonical/cloud-init/22.4.2/ChangeLog
-    - Includes (LP: #1997559, #1844191) not present in 22.4.0.
+  * d/patches/retain-netplan-world-readable.patch:
+    - Retain original world-readable perms of /etc/netplan/50-cloud-init.yaml.
+      Lunar made the config root read-only.
+  * Upstream snapshot based on 23.1. (LP: #2008230).
+    List of changes from upstream can be found at
+    https://raw.githubusercontent.com/canonical/cloud-init/23.1/ChangeLog
 
- -- James Falcon <james.falcon@canonical.com>  Mon, 28 Nov 2022 10:45:45 -0600
+ -- Alberto Contreras <alberto.contreras@canonical.com>  Thu, 23 Feb 2023 12:32:59 +0100
 
 cloud-init (22.4-0ubuntu0~22.10.1) kinetic; urgency=medium
 ```

blackboxsw replied:
I pushed https://github.com/blackboxsw/cloud-init/tree/ubuntu/kinetic  as comparative reference

blackboxsw replied:
in the case of the ChangeLog in the hotfix branches, we've already sync'd those changes into tip of main/ChangeLog so no need to pull in those diffs into the packaging branch as they get pulled in by new_upstream_snapshot.py anyway.

blackboxsw reviewed (changes requested):
Thanks @aciba90 I think all ubuntu/* package branches need to take into account the hotfix d/changelog we had already published from the related `ubuntu/<SERIES>-22.4.x` branch. Then we can new_upstream_snapshot.py after d/changelog is fixed to represent latest released d/changelog.  Same coment for the other ubuntu/* branches.

blackboxsw commented:
> 

If not the quilt references in the top of new upstream snapshot,...... here's [my `~/.quiltrc`](https://paste.ubuntu.com/p/fmZ948GQfv/)

aciba90 commented:
Thanks, @TheRealFalcon and @blackboxsw, for pointing out the missing things and the help.

About the quilt patches, I found out that `new_upstream_snapshot.py` was not discovering the patches because of a misconfiguration in my host. I have added a fix to [1] to make it host-config agnostic.

About the sync with hotfix branches, I have added the explanation to [1].

I have addressed both points in this branch. I will continue working on the others.

[1] https://github.com/canonical/uss-tableflip/pull/104

blackboxsw reviewed (approved):
Thanks @aciba90. only iff in changelog is now authorship w/ my local PR
```diff
diff --git a/debian/changelog b/debian/changelog
index 8998e833..77a3e675 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -7,7 +7,7 @@ cloud-init (23.1-0ubuntu0~22.10.1) kinetic; urgency=medium
     List of changes from upstream can be found at
     https://raw.githubusercontent.com/canonical/cloud-init/23.1/ChangeLog
 
- -- Alberto Contreras <alberto.contreras@canonical.com>  Fri, 24 Feb 2023 10:07:32 +0100
+ -- Chad Smith <chad.smith@canonical.com>  Thu, 23 Feb 2023 22:37:55 -0700
 
 cloud-init (22.4.2-0ubuntu0~22.10.1) kinetic; urgency=medium
```


sbuild and build-package look good, no missing datasource configuration 
I'll go through the remaining ubuntu/series now

aciba90 commented:
> only iff in changelog is now authorship w/ my local PR

Amended, thanks.

aciba90 commented:
Tagged and pushed.
